src/Microsoft.VisualStudio.Editor.Razor/EditorSettingsUpdateSource.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Razor.Editor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public abstract class EditorSettingsUpdateSource
+    {
+        public abstract void Update(EditorSettings updateSettings);
+    }
```

Comment:
I don't think this is better than just exposing the same API as EditorSettingsManager. Having a write-only interface is wierd